scripts/util.py

```diff
@@ -51,8 +64,77 @@ def read_source_root_path():
 def read_proof_root_path():
     return read_path_from_stdin("the 'proofs' directory (usually '.')")
 
-def read_litani_path():
-    return read_path_from_stdin("the litani script")
+def is_litani_submodule_in_path(path):
+    try:
+        sys.path.insert(1, path)
+        importlib.import_module("litani")
+        is_litani_submodule_in_path = True
+    except ModuleNotFoundError:
+        print(f"Litani submodule was not found at directory: {path})")
+        is_litani_submodule_in_path = False
+    finally:
+        del sys.path[1]
+    return is_litani_submodule_in_path
+
+def add_litani_submodule(path):
+    """Adds the litani submodule to the specified path"""
+    try:
+        subprocess.run(
+             ["git", "submodule", "add", "--quiet", "--name", "litani", "--",
+             "https://github.com/awslabs/aws-build-accumulator.git", "litani"],
+             check=True, cwd=path)
+    except subprocess.CalledProcessError:
+        print(f"Could not add Litani submodule at path: {path}")
+
+def is_litani_installed():
+    return shutil.which("litani") is not None
+
+def determine_litani_origin():
+    choice = None
+    is_litani_option_chosen = False
+    while not is_litani_option_chosen:
+        print(LITANI_DIALOG, file=sys.stderr)
+        choice = read_from_stdin()
+        if choice not in ("1", "2", "3"):
+            print("You have entered an invalid value. Please choose again.")
+        else:
+            is_litani_option_chosen = True
+    return choice
+
+def install_litani_brew_package():
+    """Runs `brew install litani`. Exits if a failure occurs"""
+    try:
+        proc = subprocess.run(
+            ["brew", "install", "litani"],
+            check=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        if proc.returncode:
+            logging.warning(proc.stderr.strip().decode())
+            sys.exit(1)
+    except subprocess.CalledProcessError as e:
+        logging.warning("Failed in running `brew install litani`")
+        logging.warning(e)
+
+def install_litani_submodule(relpath):
+    absolute_path = os.path.abspath(os.path.expanduser(relpath))
+    if not is_litani_submodule_in_path(absolute_path):
+        print(textwrap.fill(
+            "Do you want the Litani submodule to be added to the "
+            "specified path? [y/Y] "), file=sys.stderr)
```

Comment:
Usually the dialogue is `[y/N]` or `[Y/n]` I believe, with the recommended/default choice being the capital, not `[y/Y]`

Reply:
Thanks. Switched to [Y/n]. Hadn't noticed the typographical error.